### PR TITLE
Fixed broken search functionality on the client’s page and project page

### PR DESCRIPTION
**Pull request by prasanthchaduvula**
## Notion card
https://www.notion.so/Search-on-the-client-s-page-and-project-page-is-broken-a4da0ef2d0b942ce99752353dba14f4d?d=3b098a3f767f4d3fb8b63b566c86da0b
## Summary

**Before**
- Search functionality is broken on the client’s page and project page
- The page is broken when a user enters any keyword in the search input

Loom: [https://www.loom.com/share/de7d854005d9456197161a44e09cb924](https://www.loom.com/share/de7d854005d9456197161a44e09cb924)

**After**

- The user is able to search for clients using the client name on the client page search input.
- The user is able to search for projects using the project name on the project page search input.

**Code Changes**
- Removed the `AutoComplete` package and the common component built using the same name.
- Created a common component named `AutoSearch` which will take care of the search functionality along with the styling.
- Whenever we want to implement denounced search, we can simply import `AutoSearch`. Pass `searchQuery` & `searchDropdownRow` props to it.

## Preview

Screenshot 2022-12-08 at 11 06 00 PM

Screenshot 2022-12-08 at 11 06 26 PM

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (non-breaking and retains same functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

File: app/javascript/src/components/Projects/List/Header.tsx [hunk 1]
```diff
@@ -1,28 +1,17 @@
...
-  const searchCallBack = async (searchString, setDropdownItems) => {
-    try {
-      if (!searchString) return;
-      const res = await projectApi.search(searchString);
-      const dropdownList = res.data.projects.map(project => ({
-        label: `${project.name} • ${project.client_name}`,
-        value: project.id,
-      }));
+  const fetchProjects = async searchString => {
+    const res = await projectApi.search(searchString);
```
Comment (onkar-saeloun) [inline app/javascript/src/components/Projects/List/Header.tsx:12]: please wrap with try catch bock

File: app/javascript/src/components/Team/List/Header.tsx [hunk 2]
```diff
@@ -3,42 +3,16 @@ import React from "react";
 import { PlusIcon } from "miruIcons";
 
 import { TeamModalType } from "constants/index";
-// import Logger from "js-logger";
```
Comment (onkar-saeloun) [inline app/javascript/src/components/Team/List/Header.tsx:6]: do we need commented code? if no please remove

**Review by onkar-saeloun — APPROVED** [review]
Just 2 minor changes LGTM other than that

File: app/javascript/src/components/Projects/List/Header.tsx [hunk 3]
```diff
@@ -32,19 +21,7 @@ const Header = ({ setShowProjectModal, isAdminUser }) => {
...
     >
       <h2 className="header__title">Projects</h2>
-      <div className="header__searchWrap">
-        <div className="header__searchInnerWrapper">
-          <AutoComplete searchCallBack={searchCallBack} />
-          <button className="absolute inset-y-0 right-0 flex cursor-pointer items-center pr-3">
-            <SearchIcon size={12} />
-          </button>
-        </div>
-        {/* {isAdminUser && (
```
Comment (abinashpa) [inline app/javascript/src/components/Projects/List/Header.tsx:42]: Can you remove this comment?

File: app/javascript/src/common/AutoSearch/SearchDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+
+const SearchDropdown = ({ list = [], display, SearchedDataRow }) => {
```
Comment (abinashpa) [inline app/javascript/src/common/AutoSearch/SearchDropdown.tsx:3]: Can you please add an interface for the parameter?

File: app/javascript/src/common/AutoSearch/index.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from "react";
+
+import { useDebounce } from "helpers";
+import { SearchIcon, XIcon } from "miruIcons";
+
+import SearchDropdown from "./SearchDropdown";
+
+const AutoSearch = ({ searchAction, searchDataRow }) => {
```
Comment (abinashpa) [inline app/javascript/src/common/AutoSearch/index.tsx:8]: Can you please add an interface for the parameter? Name the interface Iprops.

File: app/javascript/src/components/Clients/List/Header.tsx [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
-import { SearchIcon, PlusIcon } from "miruIcons";
+import { PlusIcon } from "miruIcons";
 
 import clientApi from "apis/clients";
-import AutoComplete from "common/AutoComplete";
+import AutoSearch from "common/AutoSearch";
+
+import SearchDataRow from "./SearchDataRow";
 
 import { unmapClientListForDropdown } from "../../../mapper/client.mapper";
```
[the remark this answers is not in the document]
Comment (prasanthchaduvula) [reply app/javascript/src/components/Clients/List/Header.tsx:10]: @abinashpa Will pick it up as separate issue

File: app/javascript/src/components/Projects/List/Header.tsx (hunk 3 above)
Comment (prasanthchaduvula) [reply app/javascript/src/components/Projects/List/Header.tsx:42]: @abinashpa This is another feature I think. We can remove it after confirming it from Supriya.

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**